Lib/pickle.py

```diff
@@ -1033,31 +1033,25 @@ def _batch_appends(self, items, obj):
                 write(APPEND)
             return
 
-        it = iter(items)
         start = 0
-        while True:
-            tmp = list(islice(it, self._BATCHSIZE))
-            n = len(tmp)
-            if n > 1:
+        for batch in batched(items, self._BATCHSIZE):
```

Comment:
Is it worth using `tmp` instead of `batch`? It reduces the diff, but feel free to disregard.

Reply:
I think it's better to have a named variable that says what it is. "tmp" is too generic IMO.

Reply:
I also think `chunk` would be better, should we change it to chunk?

Reply:
Somce we use "batchsize" I think it's fine to keep "batch"